### PR TITLE
ADR-1526 private beta pay by bank transfer message

### PR DESCRIPTION
**Pull request by simondrugan16**
Ticket: https://jira.tools.tax.service.gov.uk/browse/ADR-1526

Added required warning message from confluence. Also noticed that we were passing a lot of stuff into the view so bundled that up into a view model and moved some logic into the a different model

### CODE REVIEW

File: app/controllers/checkAndSubmit/ReturnSubmittedController.scala [hunk 1]
```diff
@@ -16,63 +16,39 @@
...
-                returnDetails,
-                periodStartDate,
-                periodEndDate,
-                formattedProcessingDate,
-                formattedPaymentDueDate,
-                periodKey,
-                businessTaxAccountUrl
-              )
-            )
+            val viewModel: ReturnSubmittedViewModel = returnSubmittedHelper.getReturnSubmittedViewModel(returnDetails)
```
Comment (ajchilts) [inline app/controllers/checkAndSubmit/ReturnSubmittedController.scala:50]: Nice refactor 👍

File: app/viewmodels/returns/ReturnSubmittedHelper.scala [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      periodKey = periodKey,
+      businessTaxAccountUrl = appConfig.businessTaxAccountUrl,
+      warningText = getWarningText
+    )
+
+  }
+
+  private def getWarningText(implicit messages: Messages): WarningText =
+    WarningText(
+      iconFallbackText = Some("Warning"),
```
Comment (ajchilts) [inline app/viewmodels/returns/ReturnSubmittedHelper.scala:75]: Should this still be in the messages file so we can have Welsh translation for fallback message?

File: app/views/checkAndSubmit/ReturnSubmittedView.scala.html [hunk 3]
```diff
@@ -28,47 +27,50 @@
...
-    @paragraph(s"${messages("returnSubmitted.panel.body.dutyOwed")} ${messages(Money.format(returnDetails.amount))}", classes = "")
+    @paragraph(messages("returnSubmitted.panel.body.reference", viewModel.returnDetails.chargeReference.get.value), classes = "")
+    @paragraph(s"${messages("returnSubmitted.panel.body.dutyOwed")} ${messages(Money.format(viewModel.returnDetails.amount))}", classes = "")
     @paragraph(messages("returnSubmitted.panel.body.nil"), classes = "")
 }
 
 @layout(pageTitle = titleNoForm(messages("returnSubmitted.title")), showBackLink = false) {
 
-    @if(returnDetails.amount > 0) {
+    @if(viewModel.returnDetails.isAmountGreaterThanZero) {
```
Comment (ajchilts) [inline app/views/checkAndSubmit/ReturnSubmittedView.scala.html:52]: Nice refactor 👍

File: app/viewmodels/returns/ReturnSubmittedHelper.scala (hunk 2 above)
Comment (simondrugan16) [reply app/viewmodels/returns/ReturnSubmittedHelper.scala]: moved to messages :)

File: app/viewmodels/returns/ReturnSubmittedHelper.scala [hunk 4]
```diff
@@ -72,7 +76,7 @@ class ReturnSubmittedHelper @Inject() (dateTimeHelper: DateTimeHelper, appConfig
 
   private def getWarningText(implicit messages: Messages): WarningText =
     WarningText(
-      iconFallbackText = Some("Warning"),
+      iconFallbackText = Some(messages("returnSubmitted.warningFallbackText")),
```
Comment (ajchilts) [inline app/viewmodels/returns/ReturnSubmittedHelper.scala:79]: 👍

File: conf/messages.en [hunk 5]
```diff
@@ -956,5 +956,6 @@ returnSubmitted.direct.debit.p1 = You can pay duty automatically by direct debit
 returnSubmitted.direct.debit.setupOrManageLink.text = Set up or manage your Direct Debit
 returnSubmitted.negative.p1 = You can claim duty owed now or later in your Alcohol Duty account. If you do not claim it back then it will be used toward any duty due from your next return.
 returnSubmitted.warningText = Our bank details have changed. Choose Pay now and then Bank transfer (BACS/CHAPS) to see the new details.
+returnSubmitted.warningFallbackText = Warning
```
Comment (ajchilts) [inline conf/messages.en:959]: Do we need a Welsh translation for this also?

Comment (simondrugan16) [reply conf/messages.en:959]: We're still waiting on all the Welsh translations, so none for this yet :(

**Review by ajchilts — APPROVED** [review]

**Pull request merged**